src/coin_test/data/dataset_generator.py

```diff
@@ -27,24 +29,99 @@ def generate(
+        self,
+        dataset: "StitchedChunkDatasetGenerator.DATASET_TYPE",
+        chunk_size: int = 10,
+    ) -> None:
+        """Initialize a chunk synthetic dataset generator."""
         self.dataset = dataset
         self.start: pd.Period = dataset.df.index[0]  # type: ignore
         self.metadata = dataset.metadata
+        self.chunk_size = chunk_size
+
+        if chunk_size < 1:
+            raise ValueError("Chunk size must be a positive integer")
+        elif len(self.dataset.df) < chunk_size:
+            raise ValueError("Chunk size mustn't be larger than the dataset")
+
+    @staticmethod
+    def create_index(
+        start: pd.Period, timedelta: pd.Timedelta | pd.DateOffset, freq: str
+    ) -> pd.PeriodIndex:
+        """Create a PeriodIndex given a start time, timedelta, and frequency."""
+        return pd.period_range(start=start, end=start + timedelta, freq=freq)
+
+    @staticmethod
+    def normalize_row_data(df: pd.DataFrame) -> pd.DataFrame:
+        """Normalize the row data so that it can be sampled with returns."""
+        df = df.reset_index(drop=True)
+
+        # Normalize high, low, and close from open prices
+        df["High"] = df["High"] / df["Open"]
+        df["Low"] = df["Low"] / df["Open"]
+        df["Close_Norm"] = df["Close"] / df["Open"]
+
+        # Normalize open prices from previous close prices
+        df["Open"] /= df["Close"].shift(1, fill_value=df.iloc[0]["Open"])
+
+        # Restructure df
+        del df["Close"]
+        df.rename(columns={"Close_Norm": "Close"}, inplace=True)
+
+        return df
+
+    @staticmethod
+    def unnormalize(normalized_df: pd.DataFrame) -> pd.DataFrame:
+        """Take a normalized Dataframe and unnormalize it.
+
+        Essentially, convert columns from representing percentage increases
+        to actual prices.
+
+        Args:
+            normalized_df: Normalized Dataframe
+
+        Returns:
+            pd.DataFrame: The unnormalized Dataframe
+        """
+        # Stack Open and Close data in one column
+        oc_stacked = normalized_df.melt(
+            id_vars=["High", "Low"],
+            value_vars=["Open", "Close"],
+            var_name="O/C",
+            value_name="Perc_Change",
+            ignore_index=False,
+        ).reset_index()
+        oc_stacked.sort_values(
+            by=["index", "O/C"], ascending=[True, False], inplace=True
+        )
+
+        # Calculate Open and Close data
+        oc_stacked["Price"] = oc_stacked["Perc_Change"].cumprod()
+
+        # Unstack Open and Close data
+        oc_unstacked = oc_stacked.pivot(index="index", columns="O/C", values="Price")
+
+        # Put unstacked data in sample data
+        normalized_df["Open"] = oc_unstacked["Open"]
+        normalized_df["Close"] = oc_unstacked["Close"]
+        normalized_df["High"] = oc_unstacked["Open"] * normalized_df["High"]
+        normalized_df["Low"] = oc_unstacked["Open"] * normalized_df["Low"]
+
+        return normalized_df
```

Comment:
why are you returning a normalized df if the method is called unnormalize?

Reply:
yeah I kept the same variable name I can change that